### ISSUE TITLE
erl_tar:extract forbids any ".." relative symlink

### ISSUE THREAD

**Issue by aherranz** [erlang/otp#5765]
**Describe the bug**
If a tar archive contains a symlink starting with ".." NOT pointing outside the targeted extraction directory, erl_tar still returns {error,{Symlink,unsafe_symlink}}.

**To Reproduce**
Create a tar file with a symlink starting with ".." NOT pointing outside the targeted directory and apply erl_tar:extract.

**Expected behavior**

No error returned.

**Affected versions**

At least from 22.

**Comment by bjorng** [comment]
A path is considered unsafe if a ".." component climbs up above the extraction directory, even if it later climbs down. A symlink starting with ".." is therefore always considered unsafe.

**Comment by sheerlox** [comment]
**tl;dr**: the [`write_extracted_element`](https://github.com/erlang/otp/blob/462c6960003f6af3d7b9414f3622979b73301b6c/lib/stdlib/src/erl_tar.erl#L1645) / [`create_symlink`](https://github.com/erlang/otp/blob/462c6960003f6af3d7b9414f3622979b73301b6c/lib/stdlib/src/erl_tar.erl#L1708) functions don't seem to take into account the position of the symlink when calling [`safe_relative_path`](https://github.com/erlang/otp/blob/b2ee4fc9a0b81a139dad2033e9b2bfc178146886/lib/stdlib/src/filelib.erl#L773-L779).

I'm trying to extract [this Node.js archive](https://nodejs.org/dist/v20.10.0/node-v20.10.0-linux-x64.tar.gz) using the `erl_tar.extract` function (in Elixir) as below:
```elixir
case :erl_tar.extract({:binary, archive}, [:compressed, cwd: to_charlist(destination)]) do
  :ok -> :ok
  error -> raise "#{inspect(error)}"
end
```

And I'm getting this error:
```
{:error, {~c"../lib/node_modules/npm/bin/npm-cli.js", :unsafe_symlink}}
```

Here are the first few lines of the result of `tar -tvf node-v20.10.0-linux-x64.tar.gz` to get a better picture of its content:
![Screenshot from 2023-11-27 18-58-19](https://github.com/erlang/otp/assets/11234273/c1966e58-fbc1-432b-b73d-7cdac1d79e48)

As you can see in the screenshot above, the root of the archive contains a folder, and the `node-v20.10.0-linux-x64/bin/npm` symlink is pointing to `node-v20.10.0-linux-x64/lib/node_modules/npm/bin/npm-cli.js` because the symlink is relative its position and not the value of `cwd`.

Because this symlink never climbs up above the extraction directory, and it doesn't even climb above the root folder in the archive (`node-v20.10.0-linux-x64`), I'm expecting this to work. I gave a look at [`lib/stdlib/src/filelib.erl`](https://github.com/erlang/otp/blob/b2ee4fc9a0b81a139dad2033e9b2bfc178146886/lib/stdlib/src/filelib.erl#L773-L834), but that's more recursion logic than my brain can handle :sweat_smile: 

Erlang version: `26.1.2` (also confirmed on `24.3.4`).
